frontend/packages/gitops-plugin/src/components/list/GitOpsTableRow.tsx

```diff
@@ -3,32 +3,110 @@ import i18n from '@console/internal/i18n';
   classNames('pf-m-hidden', 'pf-m-visible-on-md', 'pf-m-width-20'), // Environments
-  classNames('pf-m-hidden', 'pf-m-visible-on-lg', 'pf-m-width-20'), // Last deployment
+  classNames('pf-m-hidden', 'pf-m-visible-on-lg', 'pf-m-width-30'), // Last deployment
 ];
 
-const handleClick = (appGroup: GitOpsAppGroupData) => {
+const handleClick = (appGroup) => {
   history.push(`/environments/${appGroup.name}?url=${appGroup.repo_url}`);
 };
 
+interface SyncProps {
+  tooltip: any[];
+  count: number;
+  icon: string;
+}
+
 const GitOpsTableRow: RowFunction<GitOpsAppGroupData> = (props) => {
   const { obj: appGroup, index, key, style } = props;
   const t = (tKey) => i18n.t(tKey);
+  let numOfSynced = 0;
+  let numOfOutOfSync = 0;
+  let numOfUnknown = 0;
+  const ttSynced = [];
+  const ttOutOfSync = [];
+  const ttUnknown = [];
+  let latestDeployedTime = '';
+  let latestDeployedEnv = '';
+  if (appGroup && appGroup.sync_status && appGroup.last_deployed) {
+    let envs = appGroup.environments.values();
+    appGroup.sync_status.forEach((status) => {
+      const env = envs.next().value; // 1:1 between a status and an env
+      if (status === 'Synced') {
+        ttSynced.push(
+          <Split hasGutter key={`${appGroup.name}-${env}`}>
+            <SplitItem>
+              <GreenCheckCircleIcon />
+            </SplitItem>
+            <SplitItem isFilled style={{ color: 'white' }}>
+              {env}
+            </SplitItem>
+            <SplitItem style={{ color: 'white' }}>{i18n.t('gitops-plugin~Synced')}</SplitItem>
+          </Split>,
+        );
+        numOfSynced++;
+      } else if (status === 'OutOfSync') {
+        ttOutOfSync.push(
+          <Split hasGutter key={`${appGroup.name}-${env}`}>
+            <SplitItem>
+              <YellowExclamationTriangleIcon />
+            </SplitItem>
+            <SplitItem isFilled style={{ color: 'white' }}>
+              {env}
+            </SplitItem>
+            <SplitItem style={{ color: 'white' }}>{i18n.t('gitops-plugin~OutOfSync')}</SplitItem>
+          </Split>,
+        );
+        numOfOutOfSync++;
+      } else if (status === 'Unknown') {
+        ttUnknown.push(
+          <Split hasGutter key={`${appGroup.name}-${env}`}>
+            <SplitItem>
+              <GrayUnknownIcon />
+            </SplitItem>
+            <SplitItem isFilled style={{ color: 'white' }}>
+              {env}
+            </SplitItem>
+            <SplitItem style={{ color: 'white' }}>{i18n.t('gitops-plugin~Unknown')}</SplitItem>
+          </Split>,
+        );
+        numOfUnknown++;
+      }
+    });
+    envs = appGroup.environments.values();
+    appGroup.last_deployed.forEach((deployedTime) => {
+      const env = envs.next().value;
+      if (latestDeployedTime < deployedTime) {
+        latestDeployedTime = deployedTime;
+        latestDeployedEnv = env;
+      }
+    });
+  }
+
+  if (latestDeployedTime === '') {
+    latestDeployedTime = null;
+  }
```

Comment:
Both of these are falsy... any reason you want `null` specifically?

Reply:
I removed this. Done.